src/python/pants/backend/docker/target_types.py

```diff
@@ -333,6 +334,55 @@ def option_values(self, **kwargs) -> Iterator[str]:
         yield from cast("tuple[str]", self.value)
 
 
+class DockerBuildUnaryOptionFieldMixin(Field):
```

Comment:
What is the motivation for not inheriting from the `DockerBuildOptionFieldMixin` class for these new option mixin classes?

I would prefer to keep the class tree cleaner and implement these new flavours by overriding the `option_values` method in new mixin classes inheriting from `DockerBuildOptionFieldMixin` if that isn't too cumbersome.

Reply:
Also, I don't really get the `unary` name? 
How about something like `DockerBuildOptionFieldValueMixin`. It isn't a perfect name either but..

Reply:
The motivation was to give different behavior to options() rather than cracking open the final method in ABC. However, I think differences can be factored out so that all build options will still live under one umbrella. I will do a follow-up PR.

Reply:
Also renaming to `DockerBuildOptionFieldValueMixin` per your suggestion